### PR TITLE
feat: hide settings email address

### PR DESCRIPTION
**Pull request by OxrxL**
https://github.com/revanced/revanced-integrations/pull/149

### CODE REVIEW

**Comment by OxrxL** [comment]
@oSumAtrIX patch code is weird like on the previous PR, but it's just a placeholder until will not be rewritten.

File: src/main/kotlin/app/revanced/patches/youtube/layout/emailaddressvisibility/patch/EmailAddressVisibilityPatch.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.builder.instruction.BuilderInstruction35c
+import org.jf.dexlib2.iface.instruction.NarrowLiteralInstruction
+import org.jf.dexlib2.iface.instruction.TwoRegisterInstruction
+import org.jf.dexlib2.iface.instruction.WideLiteralInstruction
+import org.jf.dexlib2.iface.instruction.formats.Instruction31i
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/emailaddressvisibility/patch/EmailAddressVisibilityPatch.kt:28]:
```suggestion
@DependsOn([IntegrationsPatch::class, SettingsPatch::class, ResourceIdMappingProviderResourcePatch::class])
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Overall a bit messy, would require some refactoring before merging.

**Comment by OxrxL** [comment]
@oSumAtrIX Refactored ;)

File: src/main/kotlin/app/revanced/patches/youtube/layout/personalinformation/patch/HideEmailAddressPatch.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        val setVisibilityRegister = (accountSwitcherAccessibilityLabelInstruction[setVisibilityIndex] as BuilderInstruction35c).registerD
+        val toggleRegister = (setVisibilityRegister - 1)
+
+        accountSwitcherAccessibilityLabelMethod.addInstructions(
+            setVisibilityIndex, """
+            invoke-static {}, Lapp/revanced/integrations/patches/HideEmailAddressPatch;->hideEmailAddress()Z
+            move-result v$toggleRegister
+            if-eqz v$toggleRegister, :hide
+            const/16 v$setVisibilityRegister, 0x8
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/personalinformation/patch/HideEmailAddressPatch.kt:75]: All of this should be refactored. For two registers a proper pattern should be used in the fingerprint.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Pull request merged**